client/app/components/compose.coffee

```diff
@@ -478,99 +497,67 @@ ComposeEditor = React.createClass
-                            not parent.parentNode.classList.contains 'rt-editor')
-                        after = p2
-                        inserted = document.createElement 'p'
-                        inserted.innerHTML = "<br />"
-                        if parent.nextSibling
-                            parent.parentNode.insertBefore inserted, parent.nextSibling
-                            parent.parentNode.insertBefore after, parent.nextSibling
+                        if matchesSelector?
+                            return matchesSelector.call node, '.rt-editor blockquote, .rt-editor blockquote *'
                         else
-                            parent.parentNode.appendChild inserted
-                            parent.parentNode.appendChild after
-                        ###
+                            while node? and node.tagName isnt 'BLOCKQUOTE'
+                                node = node.parentNode
+                            return node.tagName is 'BLOCKQUOTE'
+
+                    # try to get the real target element
+                    target = document.getSelection().anchorNode
+                    if target.lastChild?
+                        target = target.lastChild
+                        if target.previousElementSibling?
+                            target = target.previousElementSibling
+                    targetElement = target
+                    while targetElement and not (targetElement instanceof Element)
+                        targetElement = targetElement.parentNode
+                    if not target?
+                        return
+                    if not isInsideQuote(targetElement)
+                        # we are not inside a blockquote, nothing to do
+                        return
+
+                    # Insert another break, then select it and use outdent to
+                    # remove blocquotes
+                    if gecko
+                        br = "\r\n<br>\r\n<br class='cozyInsertedBr'>\r\n"
+                    else
+                        br = "\r\n<div></div><div><br class='cozyInsertedBr'></div>\r\n"
+                    document.execCommand 'insertHTML', false, br
 
-                        setTimeout ->
-                            inserted.focus()
-                        , 0
-                        sel = window.getSelection()
-                        sel.collapse inserted, 0
+                    if gecko
+                        node = document.querySelector('.cozyInsertedBr').previousElementSibling
+                    else
+                        node = document.querySelector('.cozyInsertedBr')
+                    getPath = (node) ->
+                        path = node.tagName
+                        while node.parentNode? and node.contentEditable isnt 'true'
+                            node = node.parentNode
+                            path = "#{node.tagName} > #{path}"
+                        return path
+                    s = window.getSelection()
```

Comment:
Don't use single letter variable it's really harmful for the maintenance (lack of meaning, hard to replace and confusing).